packages/flutter/lib/src/widgets/routes.dart

```diff
@@ -236,11 +251,59 @@ abstract class TransitionRoute<T> extends OverlayRoute<T> implements PredictiveB
   /// the transition controlled by the animation controller created by
   /// [createAnimationController()].
   Animation<double> createAnimation() {
-    assert(!_transitionCompleter.isCompleted, 'Cannot reuse a $runtimeType after disposing it.');
+    assert(!debugIsDisposed(), 'Cannot reuse a $runtimeType after disposing it.');
     assert(_controller != null);
     return _controller!.view;
   }
 
+  /// The simulation used by the most recent pushing or popping transition, if
+  /// any.
+  ///
+  /// The value is null before the first transition, or if the route does not
+  /// use simulation for transition.
+  ///
+  /// See also:
+  ///
+  ///  * [createSimulation], which creates this value.
+  Simulation? get simulation => _simulation;
```

Comment:
Seems a bit weird that we expose this property. Can we avoid this?

Reply:
We can, but I don't find it especially weird. If a route is driven by a simulation, then it's very reasonable for query the current simulation in progress. In fact, some cases might even need to the simulation object to access its status such as `isDone` for when the animation is considered completed (since it might have a very long tail) or `dx` to smoothly connect simulations.

I can remove this parameter until we really need it, if you insist. What do you think?

Reply:
I am worry about exposing transient state, this may be error prone if developer calls this at the wrong time. If this is useful in some callback, we should consider it as a pass in parameter to those call back. I vote for removing this now and expose it in the future once we have more use cases.